keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,995 @@
+resource backed by DRA. Because it cannot be scheduled after all. It does not need
+to remove the Pod from the claim.status.reservedFor field, as the special claim
+has not been created in API server yet.
+
+#### Prebind
+This is called in a separate goroutine. The plugin makes API call to create the
+`ResourceClaim` and updates the pod's status `ExtendedResourceClaimStatus`. If
+some API request fails now, PreBind fails and the pod must be retried.
+
+### Actuation for Extended Resource backed by DRA
+When a pod with extended resources requests is picked up by the kubelet on the
+node it is scheduled to run, the following are particularly important:
+
+1. Kubelet tries to admit the pod, the pod's extended resources requests
+   should not be checked against the `Node`'s allocatable, as the resources are
+   in `ResourceSlice`, not in `Node`. In reality, the current predicate.go has
+   already removed the missing extended resources from node info for
+   cluster-level resources, hence there is no extra logic needed to admit the
+   extended resources backed by DRA.
+
+1. Kubelet (DRA manager) passes the special `ResoureClaim` to DRA driver to
+   prepare the devices, in the same way as that for normal `ResourceClaim`.
+
+1. Kubelet passes the device IDs through CDI to the containers with the
+   extended resource requests. This is different from actuation of a pod with
+   resource claim, as the pod does *not* have claim requests in containers or
+   pods. Instead, the pod.status.extendedResourceClaimStatus has the mapping of
+   container name and extended resource name to request in
+   `claim.spec.devices.requests`, DRA manager uses this status information to
+   pass the proper allocated device IDs to the proper container.
+
+### Test Plan
+
+<!--
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+- `<package>`: `<date>` - `<test coverage>`
+-->
+
+Start of v1.33 development cycle (v1.33.0-alpha.x-xxx-xxxxxxxxxxxx):
+
+- `k8s.io/dynamic-resource-allocation/cel`: ??.?%
+- `k8s.io/dynamic-resource-allocation/structured`: ??.?%
+- `k8s.io/kubernetes/pkg/controller/resourceclaim`: ??.?%
+- `k8s.io/kubernetes/pkg/scheduler/framework/plugins/dynamicresources`: ??.?%
```

Comment:
Please fill this in for components that you need to modify. Can wait for 1.33.0.

Reply:
done